application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,152 @@
+package org.togetherjava.tjbot.moderation;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.GenericInteractionCreateEvent;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.nio.charset.StandardCharsets;
+import java.time.temporal.TemporalAccessor;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+/**
+ * used to log activities in the mod audit log channel
+ */
```

Comment:
The Javadoc is extremely sparse. Please expand it a bit, give a quick outline about the available and important methods and so on. Also mention whether its thread-safe or not. And please proper english. So `Used` not `used` and `.` at the end of the sentence.

Reply:
you said the javadoc needs to be consistant, so I did it this way. I'll change it.

Reply:
Sure it has to be consistent. Why did that lead you to this decision?

Reply:
I'll fix this one

Reply:
yeah, it was for the params. the description is a plain english sentence. sorry for the confusion.

Reply:
changed it.